### PR TITLE
update go-tuf and simplify TUF client code

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Summary

* Simplifies TUF client code now that we can `Init` locally without a call to the remote. Merges both `NewFromEnv` and `Initialize`.

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

File: pkg/cosign/tuf/client.go [hunk 1]
```diff
@@ -133,40 +132,63 @@ func (t *TUF) Close() error {
 	return t.local.Close()
 }
 
-func NewFromEnv(ctx context.Context) (*TUF, error) {
-	// Initializes a new TUF object from the local cache or defaults.
-	t, err := newTuf(ctx)
+func initializeTUF(ctx context.Context, embed bool, mirror string, root []byte, forceUpdate bool) (*TUF, error) {
```
Comment (haydentherapper) [inline pkg/cosign/tuf/client.go:135]: could you add a comment on the function for what each of the inputs is used for?

**Review by haydentherapper — APPROVED** [review]

File: pkg/cosign/tuf/client.go (hunk 1 above)
Comment (asraa) [reply pkg/cosign/tuf/client.go:135]: done

**Comment by asraa** [comment]
updated to fix CI

**Comment by dlorenc** [comment]
You got hit with a dependency merge that needs a quick rebase :(

**Review by dlorenc — APPROVED** [review]

**Pull request merged**